v2/cli/term/page_router.go

```diff
@@ -102,6 +115,34 @@ func (r *pageRouter) loadPage(pageName string, fn func()) {
 	}()
 }
 
+func (r *pageRouter) showPage(pageName string, focusText *tview.TextView, fn func()) {
+	// This is a critical section of code. We only want one page auto-refreshing
+	// at a time.
+	r.cancelRefreshFnMu.Lock()
+	defer r.cancelRefreshFnMu.Unlock()
+	// If any page is already auto-refreshing, stop it
+	if r.cancelRefreshFn != nil {
+		r.cancelRefreshFn()
+	}
+	// Build a new context for the auto-refresh goroutine to use
+	var ctx context.Context
+	ctx, r.cancelRefreshFn = context.WithCancel(context.Background())
+	r.ShowPage(pageName) // Bring the page into focus
+	r.app.SetFocus(focusText)
+	fn()        // Synchronously refresh the page once
+	go func() { // Start auto-refreshing
+		ticker := time.NewTicker(2 * time.Second)
+		for {
+			select {
+			case <-ticker.C:
+				fn()
+			case <-ctx.Done():
+				return
+			}
+		}
+	}()
+}
+
```

Comment:
Is the only difference between this and the existing `loadPage()` function that one hides the other pages (`r.SwitchToPage()`) and the other doesn't (`r.ShowPage()`)?

If that is the case, we should figure out how to combine those into one function so that future updates don't have to be applied in two place.

We could just add something like a `hideOtherPages bool` arg to the argument list or something.

Reply:
I was thinking of doing that, since there is a lot of repeated code. I created a whole new function just to test it out and forgot to run it by you. Gonna fix now!